### ISSUE TITLE
Type application has too many types for literal union with tuple

### ISSUE THREAD

**Issue by antonagestam** [python/mypy#12679]
#### Bug Report

False positive for type application of builtin `tuple` when using pipe-syntax unions.

#### To Reproduce

Type check this code ([mypy playground](https://mypy-play.net/?mypy=latest&python=3.10&gist=dc6a4202eb7ca9efd5b66aca7dd0e4b3)):

```python
from fractions import Fraction
from typing import Union

A = tuple[int, int] | Fraction
B = Union[tuple[int, int], Fraction]
```

#### Expected Behavior

`A` should be inferred identically as `B`, with no errors.

#### Actual Behavior

mypy claims `tuple` receives too many type arguments.

```
repro-too-many-types.py:5: error: Type application has too many types (1 expected)  [misc]
    A = tuple[int, int] | Fraction
        ^
Found 1 error in 1 file (checked 1 source file)
```

#### Your Environment

- Mypy version used: 0.942 and master@[cb6581a](https://github.com/python/mypy/commit/cb6581a8f1f41bf784eebb6a1d2c8a32fb9b43f0)
- Mypy command-line flags:
- Mypy configuration options from `mypy.ini` (and other config files):
- Python version used: 3.10.1
- Operating system and version: Macos 12.3

**Comment by JelleZijlstra** [comment]
Duplicate of #11098
